app/routes/teams/$team_id.midpoint_weather.tsx

```diff
@@ -0,0 +1,87 @@
+import { json, LoaderFunction } from "remix";
+import { kvStorageFor } from "~/db.server";
+
+type Point = {
+  lat: number;
+  lon: number
+}
+
+/* https://openweathermap.org/current#data */
+export const loader: LoaderFunction = async ({ context, params }) => {
+  if (!params.team_id) {
+    throw new Response("Not found", { status: 404 });
+  }
+  const kv = kvStorageFor(context.env);
+
+  const memberKv = kv(`team::member::${params.team_id}`);
+
+  const members = await memberKv.list();
+  if (!members) {
+    throw new Response("Not found", { status: 404 });
+  }
+
+  const midpoint: Point = midPoint(members.map((m: any) => ({ lat: m.geo.lat, lon: m.geo.lon })));
+
+  const reverseGeoURL = new URL(`
+http://api.openweathermap.org/geo/1.0/reverse?lat=${midpoint.lat}&lon=${midpoint.lon}&limit=1&appid=${context.env.OPEN_WEATHER_API_KEY}`);
+
+  const midpointGeoResponse = (await (
+    await fetch(reverseGeoURL.toString(), {
+      headers: { Accept: "application/json" },
+    })
+  ).json()) as any[];
+
+  const midpointGeo = midpointGeoResponse[0];
+  console.log("midpoint geo response: ", midpointGeoResponse[0]);
+
+  const openWeatherURL = new URL(
+    `https://api.openweathermap.org/data/2.5/weather`
+  );
+  openWeatherURL.searchParams.set("units", "imperial");
+  openWeatherURL.searchParams.set("lat", String(midpoint.lat));
+  openWeatherURL.searchParams.set("lon", String(midpoint.lon));
+  openWeatherURL.searchParams.set("appid", context.env.OPEN_WEATHER_API_KEY);
+  console.log("fetching", openWeatherURL.toString());
+  const weather = (await (
```

Comment:
I think you can run this API request and the previous one in ||

Reply:
Is this what you meant? Using `Promise.all`?